### PR TITLE
created macOS MenuBar

### PR DESCRIPTION
**Pull request by SerenityUX**

### CODE REVIEW

File: Apple/App/App-macOS.entitlements [hunk 1]
```diff
@@ -8,7 +8,7 @@
 	</array>
 	<key>com.apple.security.application-groups</key>
 	<array>
-		<string>$(APP_GROUP_IDENTIFIER)</string>
+		<string>group.com.hackclub.burrow</string>
```
Comment (conradev) [inline Apple/App/App-macOS.entitlements:11]: We likely don't want this change – `APP_GROUP_IDENTIFIER` should be `group.com.hackclub.burrow`

File: Apple/App/TunnelView.swift [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
-            case .disconnected:
-                Button("Start", action: start)
-            default:
-                EmptyView()
-            }
-        }
-        .task { await tunnel.update() }
-        .padding()
+        EmptyView()
+//        VStack {
```
Comment (conradev) [inline Apple/App/TunnelView.swift:8]: You can feel free to delete the old code – we can find it later in git

File: Apple/App/BurrowApp.swift [hunk 3]
```diff
@@ -4,14 +4,50 @@ import SwiftUI
 @main
 @MainActor
 struct BurrowApp: App {
+    //To connect to the App Delegate
+    @NSApplicationDelegateAdaptor(AppDelegate.self) var delegate
```
Comment (conradev) [inline Apple/App/BurrowApp.swift:8]: Would it be possible to fix all of the Swift lints? You should see them inside of Xcode

File: Apple/App/Menu/MenuView.swift [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        VStack {
+            HStack {
+                Text("Burrow")
+                    .fontWeight(.bold)
+
+                Spacer()
+                Toggle("", isOn: $isToggled)
+                    .toggleStyle(SwitchToggleStyle(tint: .blue))
+                    .onChange(of: isToggled) { value in
+                        if value {
```
Comment (conradev) [inline Apple/App/Menu/MenuView.swift:41]: One pattern I try to conform to when writing SwiftUI is "Model-View-ViewModel". In this case, the view is `MenuView`, the view model is `Tunnel` and the model is the underlying system APIs it configures.

In the MVVM architecture, the goal is to keep these components separate from one another for better readability. In the case of MVVM any data manipulation should go inside of the view model itself (Tunnel).

I think what we want to do here is to bring `isToggled` into `Tunnel`:

```swift
extension Tunnel {
    var isOn: Binding<Bool> {
        Binding {
            switch self.status {
            case .connected, .connecting:
                return true
            default:
                return false
            }
        } set: { newValue, _ in
            if newValue {
                try? self.start()
            } else {
                try? self.stop()
            }
        }
    }
}
```

But there are more than two states (on and off), so we might need other ways to indicate that (perhaps disabling the switch when connecting, or showing a spinning indicator instead)

File: Apple/App/Menu/MenuView.swift [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                            start()
+                        } else {
+                            stop()
+                        }
+                        print("Toggle value: \(value)")
+                    }
+            }
+            Divider()
+            switch tunnel.status {
+            case .permissionRequired:
```
Comment (conradev) [inline Apple/App/Menu/MenuView.swift:51]: It feels like the request permission portion could be moved into its own view, to keep each view small

File: Apple/Burrow.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -329,13 +341,21 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = D020F66229E4A6E5002790F6 /* NetworkExtension.xcconfig */;
 			buildSettings = {
+				DEVELOPMENT_TEAM = P6PV2R9443;
```
Comment (conradev) [inline Apple/Burrow.xcodeproj/project.pbxproj:344]: These are changes made to the Xcode project that we likely don't want – you can delete these settings in Xcode by going to the Network Extension target in Xcode, to "Build Settings", show "Levels" and "Customized" and you can see which settings you need to delete

In general, we only keep Xcode settings in `xcconfig` files

**Review by conradev — CHANGES_REQUESTED** [review]

File: Apple/App/Menu/MenuView.swift (hunk 4 above)
Comment (ma1ted) [reply Apple/App/Menu/MenuView.swift:41]: > Model-View-ViewModel
JFC 😩

**Review by ma1ted — APPROVED** [review]

**Review by conradev — CHANGES_REQUESTED** [review]
Still needs some changes before merging

**Review by conradev — APPROVED** [review]

**Pull request merged**